### PR TITLE
chore(go): update dependencies

### PR DESCRIPTION
**Pull request by evansims**
## Description

This PR updates `go.mod` and the `go.sum` dependencies lockfile to use the latest versions.

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: pkg/go/go.mod [hunk 1]
```diff
@@ -1,31 +1,31 @@
 module github.com/openfga/language/pkg/go
 
-go 1.20
+go 1.21.6
```
Comment (rhamzeh) [inline pkg/go/go.mod:3]: For the language, please keep it @ 1.20 at least until [go1.22](https://tip.golang.org/doc/go1.22) is released.

**Review by rhamzeh — CHANGES_REQUESTED** [review]

File: pkg/go/go.mod (hunk 1 above)
Comment (evansims) [reply pkg/go/go.mod:3]: Do we not need to bump this minimum to ensure security updates are applied for the runtime? `1.21.5` includes security updates for the `net/http` and `path/filepath` packages.

Comment (rhamzeh) [reply pkg/go/go.mod:3]: The CLI we are forced to by the server (we bundle it as a dependency), but we should aim to support at least the versions supported by Go for security updates (currently 1.20 + 1.21, soon to be 1.21 + 1.22). [Go Release Policy](https://go.dev/doc/devel/release#policy)

> Each major Go release is supported until there are two newer major releases. For example, Go 1.5 was supported until the Go 1.7 release, and Go 1.6 was supported until the Go 1.8 release. We fix critical problems, including [critical security problems](https://go.dev/security), in supported releases as needed by issuing minor revisions (for example, Go 1.6.1, Go 1.6.2, and so on).

Comment (rhamzeh) [reply pkg/go/go.mod:3]: The fixes in go 1.21.5 were included in [1.20.12](https://github.com/golang/go/issues?q=milestone%3AGo1.20.12+label%3ACherryPickApproved)

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**